sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpError.cs

```diff
@@ -40,19 +30,194 @@ internal static class AmqpError
+                                                 string description,
+                                                 string eventHubsResource)
+        {
+            // The request timed out.
+
+            if (String.Equals(condition, TimeoutError.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new EventHubsTimeoutException(eventHubsResource, description);
+            }
+
+            // The Event Hubs service was busy.
+
+            if (String.Equals(condition, ServerBusyError.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new ServiceBusyException(eventHubsResource, description);
+            }
+
+            // An argument was rejected by the Event Hubs service.
+
+            if (String.Equals(condition, ArgumentError.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new ArgumentException(description);
+            }
+
+            if (String.Equals(condition, ArgumentOutOfRangeError.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new ArgumentOutOfRangeException(description);
+            }
+
+            // The consumer was superseded by one with a higher owner level.
+
+            if (String.Equals(condition, AmqpErrorCode.Stolen.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new ConsumerDisconnectedException(eventHubsResource, description);
+            }
+
+            // Authorization was denied.
+
+            if (String.Equals(condition, AmqpErrorCode.UnauthorizedAccess.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new UnauthorizedAccessException(description);
+            }
+
+            // Requests are being throttled due to exceeding the service quota.
+
+            if (String.Equals(condition, AmqpErrorCode.ResourceLimitExceeded.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new QuotaExceededException(eventHubsResource, description);
+            }
+
+            // The service does not understand how to process the request.
+
+            if (String.Equals(condition, AmqpErrorCode.NotAllowed.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new InvalidOperationException(description);
+            }
+
+            if (String.Equals(condition, AmqpErrorCode.NotImplemented.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new NotSupportedException(description);
+            }
+
+            // The Event Hubs resource was not valid or communication with the service was interrupted.
+
+            if (String.Equals(condition, AmqpErrorCode.NotFound.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                if (NotFoundExpression.IsMatch(description)
```

Comment:
I kept this logic to retain the same error behavior as track one.   Upon inspection, it seemed useful to keep the distinction.